core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -329,46 +329,88 @@ func (p *logEventProvider) readLogs(ctx context.Context, latest int64, entries [
 		if len(entry.addr) == 0 {
 			continue
 		}
-		// lggr := mainLggr.With("upkeep", entry.upkeepID.String(), "addrs", entry.addr, "sigs", entry.topics)
+
+		// start should either be the last block polled for the entry or the
+		// lookback range in the case this is the first time the entry is polled
 		start := entry.lastPollBlock
 		if start == 0 || start < latest-logBlocksLookback {
 			// long range or first time polling,
 			// using a larger lookback and burst
 			start = latest - logBlocksLookback*2
+
+			// start should not be negative in the special case of an empty or
```

Comment:
good catch, but it should be outside of this if statement, right before we reserve with `latest-start`

Reply:
Makes sense.